portal-web/docroot/html/taglib/ui/input_date/page.jsp

```diff
@@ -58,11 +58,12 @@ if (!BrowserSnifferUtil.isMobile(request)) {
 
 	simpleDateFormatPattern = shortDateFormatSimpleDateFormat.toPattern();
 
-	simpleDateFormatPattern = simpleDateFormatPattern.replaceAll("yyyy", "yy");
+	simpleDateFormatPattern = simpleDateFormatPattern.replaceAll("yyyy", "y");
+	simpleDateFormatPattern = simpleDateFormatPattern.replaceAll("yy", "y");
 	simpleDateFormatPattern = simpleDateFormatPattern.replaceAll("MM", "M");
 	simpleDateFormatPattern = simpleDateFormatPattern.replaceAll("dd", "d");
 
-	simpleDateFormatPattern = simpleDateFormatPattern.replaceAll("yy", "yyyy");
+	simpleDateFormatPattern = simpleDateFormatPattern.replaceAll("y", "yyyy");
```

Comment:
At this point, we already ensure that we have `y`, `M` and `d`. Why do we need to convert it back to the long format? Why can't we simply generate the mask based on that?

Reply:
Because we use that same variable (`simpleDateFormatPattern`) below, so we must leave it in a proper state.

Reply:
Thanks, didn't see past the diff 😂